pkg/capabilities/capabilities.go

```diff
@@ -105,6 +105,28 @@ type CallbackExecutable interface {
 	Execute(ctx context.Context, callback chan<- CapabilityResponse, request CapabilityRequest) error
 }
 
+type Validatable interface {
+	// GetConfigJsonSchema will create a values-wrapped JSON schema for the "config" field of the capability.
+	//
+	// This allows for more comprehensive validation of a workflow yaml file since
+	// after dynamically resolving the capability, the workflow engine can validate
+	// the config field against the schema.
+	//
+	// NOTE: What to do with method calls like:
+	//
+	//  - `RegisterToWorkflow` and `UnregisterFromWorkflow` 
+	//  - `RegisterTrigger` and `UnregisterTrigger`
+	//
+	// They take in a request object that has a `Config` field that may conflate
+	// with the `Config` field within `Execute`.
+	GetRequestConfigJsonSchema() (*CapabilityResponse)
+	// ValidateRequestConfig(config *values.Map) error
+	// ValidateRequestInput(inputs *values.Map) error
+	// NOTE: Should we need this? Or should we just document this within
+	// the capability itself?
+	// ExampleOutput() *values.Map
```

Comment:
We need this to validate input/output compatibility. We should be able to do something like
```
SrcCapability.Validate(DestCapability.ExampleOutput())
```

The above can catch some basic mismatches between something like
```
consensus:
  - type: offchain_reporting@1
    ref: evm_median
    inputs:
      observations:
        - triggers.report_data.outputs
    config:
      aggregation_method: data_feeds_2_0
      aggregation_config:
        123: # ETHUSD
          deviation: "0.005"
          heartbeat: 24h
        456: # LINKUSD
          deviation: "0.001"
          heartbeat: 24h
        789: # USDBTC
          deviation: "0.002"
          heartbeat: 6h
      encoder: EVM
      encoder_config:
          abi: "mercury_reports bytes[]"

targets:
  - type: write_polygon_mainnet@1
    inputs:
      report:
        - consensus.evm_median.outputs # SHOULD BE consensus.evm_median.outputs.report
    config:
      address: "0xaabbcc"
      method: "updateFeedValues(report bytes, role uint8)"
      params: [$(inputs.report), 1] # THIS IS AN ERROR, BECAUSE REPORT IS report.report
```

Reply:
Do we expect this to be a common use case? For some reason I thought that was an exception rather than the norm. I was thinking that often when we're chaining capabilities we might be passing in some other inputs too.